target/riscv/op_helper_cheri.c

```diff
@@ -213,12 +213,34 @@ void HELPER(cspecialrw)(CPUArchState *env, uint32_t cd, uint32_t cs,
                 scr_info[index].name, PRINT_CAP_ARGS(&new_val));
         }
 #endif
-        if (index == CheriSCR_DDC && !new_val.cr_tag)
-            qemu_log_instr_or_mask_msg(env, CPU_LOG_INT,
-                "  Note: Installed untagged $ddc at " TARGET_FMT_lx "\n",
-                cpu_get_recent_pc(env));
-        *scr = new_val;
-        cheri_log_instr_changed_capreg(env, scr_info[index].name, scr);
+        switch (index) {
+        case CheriSCR_UTCC:
+        case CheriSCR_STCC:
+        case CheriSCR_MTCC: {
+            target_ulong new_tvec = SCR_TO_PROGRAM_COUNTER(env, &new_val);
+            target_ulong new_mode = new_tvec & 3;
+            /* The low two bits encode the mode, but only 0 and 1 are valid. */
```

Comment:
What mode does this refer to?  It doesn't seem to be present in the old code?

Reply:
RISC-V's xtvec; 0 is direct, 1 is vectored, 2 is reserved but being claimed by extremely dodgy CLIC thing ("I can't believe it's not Arm-M"), 3 is reserved